packages/sdk/src/utils/verifyClient.ts

```diff
@@ -3,61 +3,83 @@
-
-    // Check protocol version.
-    if (!isSupportedProtocolVersion(protocolVersion)) {
-        // In a perfect world we would return a 403 (Forbidden) response, but due to a shortcoming in
-        // C# ClientWebSocket, we have no way to convey error details in the HTTP response:
-        //     "ClientWebSocket does not provide upgrade request error details"
-        //      https://github.com/dotnet/corefx/issues/29163
-        // As a workaround, we inject an unexpected subprotocol, resulting in an "unsupported protocol"
-        // error on the client.
-        req.headers['sec-websocket-protocol'] = 'ERR_UNSUPPORTED_PROTOCOL';
+    // Temporary: Support old clients reporting the legacy protocol version.
+    // TODO: Remove this after a few releases.
+    const legacyProtocolVersion = decodeURIComponent(headers[Constants.HTTPHeaders.LegacyProtocolVersion]);
+    if (legacyProtocolVersion === '1') {
+        return cb(true);
     }
 
-    // Client looks valid to connect.
-    return cb(true);
-}
+    // Verify minimum supported versions are met (client and SDK).
 
-/**
- * @hidden
- * Utility that checks the given protocol version against the supported version.
- * @param protocolVersion The protocol version to check.
- * @returns `true` if this is a supported protocol version. `false` otherwise.
- */
-export function isSupportedProtocolVersion(protocolVersion: number | string | null | undefined): boolean {
-    if (typeof protocolVersion === 'string') {
-        protocolVersion = Number.parseInt(protocolVersion, 10);
+    /*
+     * Due to a shortcoming in C# ClientWebSocket, we have no way to convey any error details in the HTTP response,
+     * including error code.
+     *    See: "ClientWebSocket does not provide upgrade request error details"
+     *    https://github.com/dotnet/corefx/issues/29163
+     */
+
+    // tslint:disable-next-line:variable-name
+    const CurrentClientVersion
+        = semver.coerce(decodeURIComponent(headers[Constants.HTTPHeaders.CurrentClientVersion]));
+    // tslint:disable-next-line:variable-name
+    const MinimumSupportedSDKVersion
+        = semver.coerce(decodeURIComponent(headers[Constants.HTTPHeaders.MinimumSupportedSDKVersion]));
+
+    // Test the current client version. Is it greater than or equal to the minimum client version?
+    const clientPass = semver.gte(CurrentClientVersion, MinimumSupportedClientVersion);
+    // Test the current SDK version. Is it greater than or equal to the minimum SDK version?
+    const sdkPass = semver.gte(CurrentSDKVersion, MinimumSupportedSDKVersion);
+
+    if (!clientPass) {
+        // tslint:disable-next-line:max-line-length
+        const message = `Connection rejected due to out of date client. Client version: ${CurrentClientVersion.toString()}. Min supported version by SDK: ${MinimumSupportedClientVersion.toString()}`;
+        log.info('network', message);

```

Comment:
>log.info('network' [](start = 8, length = 18)

I think these messages should be exposed to the app developer by default, not hidden with network traffic. Also, probably an `error` or `warning`, not an `info` message.

Reply:
True, they shouldn't be in with the network traffic.